crates/stackable-operator/src/commons/rbac.rs

```diff
@@ -28,24 +28,18 @@ pub enum Error {
 }
 
 /// Build RBAC objects for the product workloads.
-/// The `product_name` is meant to be the product name, for example: zookeeper, airflow, etc.
-/// and it is a assumed that a ClusterRole named `{product_name}-clusterrole` exists.
+/// The names of the service account and role binding match the following templates:
+/// - `{resource_name}-serviceaccount`
+/// - `{resource_name}-rolebinding`
+///
+/// The service account is bound to a cluster role named `{product_name}-clusterrole` which
+/// must already exist.
 pub fn build_rbac_resources<T: Clone + Resource<DynamicType = ()>>(
     resource: &T,
-    // 'product_name' is not used to build the names of the serviceAccount and roleBinding objects,
-    // as this caused problems with multiple clusters of the same product within the same namespace
-    // see <https://stackable.atlassian.net/browse/SUP-148> for more details.
-    // Instead the names for these objects are created by reading the name from the cluster object
-    // and appending [-rolebinding|-serviceaccount] to create unique names instead of using the
-    // same objects for multiple clusters.
```

Comment:
I think this comment still makes sense tbh (as internal motivation for why things are the way they are).